src/translator/output_printer.h

```diff
@@ -99,11 +99,15 @@ class OutputPrinter {
   std::string getWordScores(const Hypothesis::PtrType& hyp);
 
   float getAlignmentThreshold(const std::string& str) {
+#if WITHOUT_EXCEPTIONS
+    if (str.begin() == str.end()) return 0.f;
+#else
```

Comment:
You can remove pre-processor here. The logic of `if (str.begin() == str.end())` applies for exception friendly code as well: there is no need to throw exception if you know string is empty. As a rule a throw is always times slower than just if-check. Also, you will need that `std::stof` in the exception-free code.

Reply:
or `return str.begin() == str.end() ? 0.f : std::max(std::stof(str), 0.f);`